src/Serilog.Sinks.EventLog/Sinks/EventLog/EventLogSink.cs

```diff
@@ -53,69 +58,153 @@ public EventLogSink(string source, string logName, ITextFormatter textFormatter,
-		/// <remarks>
-		/// <see cref="LogEventLevel.Debug" />, <see cref="LogEventLevel.Information" /> and <see cref="LogEventLevel.Verbose" /> are registered as <see cref="EventLogEntryType.Information" />.
-		/// <see cref="LogEventLevel.Error" />, <see cref="LogEventLevel.Fatal" /> are registered as <see cref="EventLogEntryType.Error" />.
-		/// <see cref="LogEventLevel.Warning" /> are registered as <see cref="EventLogEntryType.Warning" />.
-		/// The Event ID in the Windows log will be set to the integer value of the <paramref name="logEvent"/>'s <see cref="LogEvent.Level"/> property, so that the log can be filtered with more granularity.</remarks>
-		public void Emit(LogEvent logEvent)
-		{
-			if (logEvent == null) throw new ArgumentNullException("logEvent");
-
-			EventLogEntryType type;
-			switch (logEvent.Level)
-			{
-				case LogEventLevel.Debug:
-				case LogEventLevel.Information:
-				case LogEventLevel.Verbose:
-					type = EventLogEntryType.Information;
-					break;
-
-				case LogEventLevel.Error:
-				case LogEventLevel.Fatal:
-					type = EventLogEntryType.Error;
-					break;
-
-				case LogEventLevel.Warning:
-					type = EventLogEntryType.Warning;
-					break;
-
-				default:
-					SelfLog.WriteLine("Unexpected logging level, writing to EventLog as Information");
-					type = EventLogEntryType.Information;
-					break;
-			}
-
-			var payloadWriter = new StringWriter();
-			_textFormatter.Format(logEvent, payloadWriter);
-
-
-            //The payload is limitted in length and allowed chars
-            //see: https://msdn.microsoft.com/en-us/library/e29k5ebc%28v=vs.110%29.aspx
-            var payload = payloadWriter.ToString();
+        /// Returns name of EventLog that contains the registered source
+        /// </summary>
+        /// <param name="source"></param>
+        /// <param name="machineName"></param>
+        /// <returns></returns>
+	    private System.Diagnostics.EventLog GetLogFromSource(string source, string machineName)
+	    {
+	        return System.Diagnostics.EventLog.GetEventLogs().ToList().FirstOrDefault(l => SourceExistsInLog(source, l.Log, machineName));
+	    }
+
+	    private bool SourceExistsInLog(string sourceName, string logName, string machineName)
+	    {
+            return FindSourceNamesFromLog(logName, machineName)?.Any(s => s == sourceName) ?? false;
+	    }
+
+        /// <summary>
+        /// Returns list of sources associated with specific log
+        /// </summary>
+        /// <remarks>
+        /// EventLog Api doesn't support this natively so we have to use the Registry
+        /// </remarks>
+        private static List<string> FindSourceNamesFromLog(string logName, string machineName)
+        {
+            if (machineName.Trim() == ".") machineName = Environment.MachineName;
+
+            var hive = RegistryKey.OpenRemoteBaseKey(RegistryHive.LocalMachine, machineName);
+            var keyLog = hive.OpenSubKey($@"SYSTEM\CurrentControlSet\Services\Eventlog\{logName}");
```

Comment:
This is the same key that is used in the `CreateEventSource` call to open a log
http://referencesource.microsoft.com/#System/services/monitoring/system/diagnosticts/EventLog.cs,447 - (and then the specific log [here](http://referencesource.microsoft.com/#System/services/monitoring/system/diagnosticts/EventLog.cs,459))

Good point though, I am assuming that the app requesting a specific log also has access to said log.  I see two options here:

1.  I can wrap this and rethrow with some details (i.e. parameters `logName`/`machineName`) 
2.  I can fall back to the Application Log and write the message there + a warning about why we failed to write to the custom log  
  1.  My concern here is that we still need a `source` for the Application log and if we don't have permissions to a custom log, I'm not sure we'd have permission to create an event source in an existing log...